src/Widgets/WindowClone.vala

```diff
@@ -295,53 +295,151 @@ namespace Gala {
-            set_size (outer_rect.width, outer_rect.height);
+            var target_x = outer_rect.x - offset_x;
+            var target_y = outer_rect.y - offset_y;
 
-            if (should_fade ())
-                opacity = 0;
+            GestureAnimationDirector.OnBegin on_animation_begin = () => {
+                window_icon.set_easing_duration (0);
+            };
 
-            restore_easing_state ();
+            GestureAnimationDirector.OnUpdate on_animation_update = (percentage) => {
+                var x = GestureAnimationDirector.animation_value (initial_x, target_x, percentage);
+                var y = GestureAnimationDirector.animation_value (initial_y, target_y, percentage);
+                var width = GestureAnimationDirector.animation_value (initial_width, outer_rect.width, percentage);
+                var height = GestureAnimationDirector.animation_value (initial_height, outer_rect.height, percentage);
+                var opacity = GestureAnimationDirector.animation_value (255f, 0f, percentage);
+
+                set_size (width, height);
+                set_position (x, y);
+                window_icon.opacity = (uint) opacity;
+                window_icon.set_position ((width - WINDOW_ICON_SIZE) / 2,
+                    height - (WINDOW_ICON_SIZE * scale_factor) * 0.75f);
+            };
 
-            if (animate)
-                toggle_shadow (false);
+            GestureAnimationDirector.OnEnd on_animation_end = (percentage, cancel_action) => {
+                window_icon.set_easing_duration (MultitaskingView.ANIMATION_DURATION);
 
-            window_icon.set_position ((outer_rect.width - WINDOW_ICON_SIZE) / 2, outer_rect.height - (WINDOW_ICON_SIZE * scale_factor) * 0.75f);
-            window_icon.opacity = 0;
-            close_button.opacity = 0;
+                if (cancel_action) {
+                    return;
+                }
+
+                save_easing_state ();
+                set_easing_mode (MultitaskingView.ANIMATION_MODE);
+                set_easing_duration (animate ? MultitaskingView.ANIMATION_DURATION : 0);
+    
+                set_position (target_x, target_y);
+                set_size (outer_rect.width, outer_rect.height);
+    
+                if (should_fade ()) {
+                    opacity = 0;
+                }
+
+                restore_easing_state ();
+    
+                if (animate) {
+                    toggle_shadow (false);
+                }
+    
+                window_icon.set_position ((outer_rect.width - WINDOW_ICON_SIZE) / 2, outer_rect.height - (WINDOW_ICON_SIZE * scale_factor) * 0.75f);
+                window_icon.opacity = 0;
+                close_button.opacity = 0;
+            };
+
+            if (gesture_animation_director == null) {
+                on_animation_begin (0);
+                on_animation_end (100, false);
+            } else {
+                gesture_animation_director.on_animation_begin.connect ((percentage) => {
```

Comment:
I feel like these could be put on one line to keep it clear:
`gesture_animation_director.on_animation_begin.connect ((percentage) => on_animation_begin (percentage))`

or simply
`gesture_animation_director.on_animation_begin.connect (on_animation_begin);`
if that works.

Reply:
The second option prints a warning:
```
 warning: Connecting delegates to signals is experimental
```

And the first one will pass the 120 characters limit. I like the first option though, but I don't know how hard is the 120 characters limit....

Reply:
@JoseExposito not that hard :) Clarity is what's important here

Reply:
Good to know. I mean, it makes a lot of sense, specially now that monitors are quiet big.

I'm going to close this PR, as the other one includes every change and the code is cleaner.

Btw, I changed this code to use a simpler method:
https://github.com/elementary/gala/pull/983/files#diff-82d4c23d6a3129070cf40cd4e43dbb42eb991ab0fe7d1bd508c7ee41a793fd60R702